### ISSUE TITLE
Path hits `is_after(to, self.current_position)` assert

### ISSUE THREAD

**Issue by Herschel** [nical/lyon#518]
The following path hit an assert in the fill tessellator in both debug and release:
```
    let mut builder = Path::builder();
    builder.move_to(Point::new(-76.95f32, -461.8f32));
    builder.quadratic_bezier_to(
        Point::new(-75.95f32, -462.6f32),
        Point::new(-74.65f32, -462.8f32),
    );
    builder.line_to(Point::new(-79.1f32, -456.4f32));
    builder.line_to(Point::new(-83.4f32, -464.75f32));
    builder.line_to(Point::new(-80.75f32, -464.75f32));
    builder.line_to(Point::new(-79.05f32, -458.1f32));
    builder.quadratic_bezier_to(
        Point::new(-78.65f32, -460.2f32),
        Point::new(-77.35f32, -461.45f32),
    );
    builder.line_to(Point::new(-77.1f32, -461.65f32));
    builder.line_to(Point::new(-76.95f32, -461.8f32));
    builder.close();
    builder.build()
```

Debug:
```
thread 'main' panicked at 'assertion failed: is_after(to, self.current_position)', ...\github.com-1ecc6299db9ec823\lyon_tessellation-0.15.1\src\fill.rs:715:17
```

Release:
```
thread 'main' panicked at 'assertion failed: from != edge.to', ...\lyon_tessellation-0.15.1\src\fill.rs:1252:13
```

SVG of above broken minimal example and full art (although the SVGs may or may not expose the problem, as Flash can sometimes export paths in awkward ways):
[bad-art.zip](https://github.com/nical/lyon/files/4006264/bad-art.zip)

lyon and lyon-tessellation 0.15.1

**Comment by Herschel** [comment]
The above example will work successfully if I omit the last `line_to` and let it close via `close`, but here's another example where omitting it has no effect:
```
    let mut builder = Path::builder();
    builder.move_to(point(-69.1f32, -465.5f32));
    builder.line_to(point(-69.1f32, -461.65f32));
    builder.quadratic_bezier_to(point(-70.95f32, -462.8f32), point(-72.95f32, -462.9f32));
    builder.quadratic_bezier_to(point(-75.65f32, -463.1f32), point(-77.35f32, -461.45f32));
    builder.quadratic_bezier_to(point(-78.65f32, -460.2f32), point(-79.05f32, -458.1f32));
    builder.line_to(point(-80.55f32, -465.5f32));
    builder.line_to(point(-69.1f32, -465.5f32)); // Still asserts when commented out
    builder.close();
```

**Comment by nical** [comment]
Thanks again for filing these issues. The minimal test cases you provide are a pleasure to work with.

**Comment by nical** [comment]
Fixed in lyon_tessellation 0.15.2.
